rest/server.go

```diff
@@ -401,15 +407,11 @@ func (server *HTTPServer) StartHTTP() error {
 		server.httpServer.Addr = bindAddr
 	}
 
-	httpHandler := server.NewMux()
-
-	if server.httpConfig.GetAllowProfiling() {
-		if httpHandler, err = xhttp.NewRequestProfiler(httpHandler, server.httpConfig.GetProfilerDir(), nil, xhttp.LogProfile()); err != nil {
-			return errors.Trace(err)
-		}
+	if server.handler == nil {
+		server.handler = server.NewMux()
 	}
 
-	server.httpServer.Handler = httpHandler
+	server.httpServer.Handler = server.handler
```

Comment:
what is the point to introduce `server.handler` if  `server.httpServer.Handler` already servers the purpose?

Reply:
because it is not public; i cannot set `server.httpServer.handler` until the http server is started. i would like the caller to be able to define a handler before calling `StartHTTP`